src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Security/DataSetDataTableInObjectGraphTests.cs

```diff
@@ -0,0 +1,424 @@
+using System;
+using System.Data;
+using Newtonsoft.Json;
+
+namespace Blah
+{
+    public class BlahClass
+    {
+        [JsonIgnore]
+        public DataTable DT;
+
+        public BlahClass Method(JsonReader reader)
+        {
+            JsonSerializer js = new JsonSerializer();
+            return (BlahClass) js.Deserialize(reader);
+        }
+    }
+}");
+        }
+
+        [Fact]
+        public async Task Newtonsoft_JsonConvert_DeserializeObject_Generic_Diagnostic()
+        {
+            await VerifyCSharpNewtonsoftAsync(@"
+using System;
+using System.Data;
+using Newtonsoft.Json;
+
+namespace Blah
+{
+    public class BlahClass
+    {
+        public DataTable DT { get; set; }
+
+        public BlahClass Method(string s)
+        {
+            return JsonConvert.DeserializeObject<BlahClass>(s);
+        }
+    }
+}",
+                GetCSharpResultAt(14, 20, "DataTable", "DataTable BlahClass.DT"));
+        }
+
+        private static async Task VerifyCSharpAsync(string source, params DiagnosticResult[] expected)
+        {
+            var csharpTest = new VerifyCS.Test
+            {
+                ReferenceAssemblies = AdditionalMetadataReferences
+                    .DefaultWithSerialization,
+                TestState =
+                {
+                    Sources = { source },
+                }
+            };
+
+            csharpTest.ExpectedDiagnostics.AddRange(expected);
+
+            await csharpTest.RunAsync();
+        }
+
+        /// <summary>
+        /// Tests code using JavaScriptSerializer.
+        /// </summary>
+        /// <param name="source">Source code to test.</param>
+        /// <param name="expected">Expected diagnostics.</param>
+        /// <returns>Task of the test run.</returns>
+        private static async Task VerifyCSharpJssAsync(string source, params DiagnosticResult[] expected)
+        {
+            var csharpTest = new VerifyCS.Test
+            {
+                ReferenceAssemblies = AdditionalMetadataReferences
+                    .DefaultWithSystemWeb
+                    .AddAssemblies(ImmutableArray.Create("System.Data")),
+                TestState =
+                {
+                    Sources = { source },
+                }
+            };
+
+            csharpTest.ExpectedDiagnostics.AddRange(expected);
+
+            await csharpTest.RunAsync();
+        }
+
+        /// <summary>
+        /// Tests code using Newtonsoft Json.NET.
+        /// </summary>
+        /// <param name="source">Source code to test.</param>
+        /// <param name="expected">Expected diagnostics.</param>
+        /// <returns>Task of the test run.</returns>
+        private static async Task VerifyCSharpNewtonsoftAsync(string source, params DiagnosticResult[] expected)
+        {
+            var csharpTest = new VerifyCS.Test
+            {
+                ReferenceAssemblies = AdditionalMetadataReferences
+                    .DefaultWithNewtonsoftJson12
```

Comment:
@dotpaul This requires the update for the method to take the NewtonsoftJson version.